tvplayer/src/main/java/com/kaltura/tvplayer/PKPlaylistController.java

```diff
@@ -0,0 +1,716 @@
+package com.kaltura.tvplayer;
+
+import android.text.TextUtils;
+
+import com.kaltura.netkit.utils.ErrorElement;
+import com.kaltura.playkit.PKError;
+import com.kaltura.playkit.PKLog;
+import com.kaltura.playkit.PKMediaEntry;
+import com.kaltura.playkit.PKPlaylist;
+import com.kaltura.playkit.PKPlaylistMedia;
+import com.kaltura.playkit.PlayerEvent;
+import com.kaltura.playkit.providers.ott.OTTMediaAsset;
+import com.kaltura.playkit.providers.ovp.OVPMediaAsset;
+import com.kaltura.tvplayer.KalturaPlayer;
+import com.kaltura.tvplayer.OTTMediaOptions;
+import com.kaltura.tvplayer.OVPMediaOptions;
+import com.kaltura.tvplayer.playlist.BasicMediaOptions;
+import com.kaltura.tvplayer.playlist.BasicPlaylistOptions;
+import com.kaltura.tvplayer.playlist.CountDownOptions;
+import com.kaltura.tvplayer.playlist.OTTPlaylistOptions;
+import com.kaltura.tvplayer.playlist.OVPPlaylistIdOptions;
+import com.kaltura.tvplayer.playlist.OVPPlaylistOptions;
+import com.kaltura.tvplayer.playlist.PKBasicPlaylist;
+import com.kaltura.tvplayer.playlist.PKPlaylistType;
+import com.kaltura.tvplayer.playlist.PlaylistController;
+import com.kaltura.tvplayer.playlist.PlaylistEvent;
+import com.kaltura.tvplayer.playlist.PlaylistOptions;
+
+import java.util.ArrayList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class PKPlaylistController implements PlaylistController {
+
+    private static final PKLog log = PKLog.get("PlaylistController");
+
+    private KalturaPlayer kalturaPlayer;
+    private PKPlaylist playlist;
+    private PKPlaylistType playlistType;
+    private PlaylistOptions playlistOptions;
+    private CountDownOptions countDownOptions;
+
+    private int currentPlayingIndex = -1;
+    private boolean playlistAutoContinue = true;
+    private boolean loopEnabled;
+    private boolean shuffleEnabled;
+    private boolean recoverOnError;
+
+    private List<PKPlaylistMedia> origlPlaylistEntries;
+    private Map<String, PKMediaEntry> loadedMediasMap; // map of the media id and it's PKMediaEntry (ovp/ott in entryId format basic any string that ws given by user as id)
+
+    private enum CacheMediaType {
+        Prev, Current, Next
+    }
+
+    public PKPlaylistController(KalturaPlayer kalturaPlayer, PKPlaylist playlist, PKPlaylistType playlistType) {
```

Comment:
Check if this playlistType parameter is needed

Reply:
yes it is used internally only